drivers/sensor/tmp108/tmp108.c

```diff
@@ -0,0 +1,406 @@
+			mode = TI_TMP108_FREQ_16_HZ;
+		}
+		result = tmp108_write_config(dev,
+				TI_TMP108_FREQ_MASK,
+				mode);
+		break;
+
+	case SENSOR_ATTR_SHUTDOWN_MODE:
+		result = tmp108_write_config(dev,
+				TI_TMP108_MODE_MASK,
+				TI_TMP108_MODE_SHUTDOWN);
+		drv_data->one_shot_mode = false;
+		break;
+
+	case SENSOR_ATTR_CONTINUOUS_CONVERSION_MODE:
+		result = tmp108_write_config(dev,
+				TI_TMP108_MODE_MASK,
+				TI_TMP108_MODE_CONTINUOUS);
+		drv_data->one_shot_mode = false;
+		break;
+
+	case SENSOR_ATTR_ONE_SHOT_MODE:
+		result = tmp108_write_config(dev,
+				TI_TMP108_MODE_MASK,
+				TI_TMP108_MODE_ONE_SHOT);
+		drv_data->one_shot_mode = true;
+		break;
+
+	case SENSOR_ATTR_ALERT_POLARITY:
+		if (val->val1 == 1) {
+			mode = TI_TMP108_CONF_POL_HIGH;
+		} else {
+			mode = TI_TMP108_CONF_POL_LOW;
+		}
+		result = tmp108_write_config(dev,
+					TI_TMP108_CONF_POL_MASK,
+					mode);
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	if(result < 0) {
+		return result;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api tmp108_driver_api = {
+	.attr_set = tmp108_attr_set,
+	.attr_get = tmp108_attr_get,
+	.sample_fetch = tmp108_sample_fetch,
+	.channel_get = tmp108_channel_get,
+	.trigger_set = tmp_108_trigger_set,
+};
+
+#ifdef CONFIG_TMP108_ALERT_INTERRUPTS
+static int setup_interrupts(const struct device *dev)
+{
+	struct tmp108_data *drv_data = dev->data;
+	const struct tmp108_config *config = dev->config;
+	const struct gpio_dt_spec *alert_gpio = &config->alert_gpio;
+	int result = 0;
+
+	if (!device_is_ready(alert_gpio->port)) {
+		LOG_ERR("tmp108: gpio controller %s not ready", 
+			alert_gpio->port->name);
+		return -ENODEV;
+	}
+
+	result = gpio_pin_configure_dt(alert_gpio, GPIO_INPUT);
+
+	if(result < 0) {
+		return result;
+	}
+
+	gpio_init_callback(&drv_data->temp_alert_gpio_cb,
+			   tmp108_trigger_handle_alert,
+			   BIT(alert_gpio->pin));
+
+	result = gpio_add_callback(alert_gpio->port,
+			  &drv_data->temp_alert_gpio_cb);
+
+	if(result < 0) {
+		return result;
+	}
+
+	result = gpio_pin_interrupt_configure_dt(alert_gpio,
+					GPIO_INT_EDGE_BOTH);
+
+	if(result < 0) {
+		return result;
+	}
+
+	return 0;
+}
+#endif
+
+static int tmp108_init(const struct device *dev)
+{
+	const struct tmp108_config *cfg = dev->config;
+	struct tmp108_data *drv_data = dev->data;
+
+	if (!device_is_ready(cfg->i2c_spec.bus)) {
+		LOG_ERR("I2C dev %s not ready", cfg->i2c_spec.bus->name);
+		return -ENODEV;
+	}
+
+	drv_data->scheduled_work.work.handler = tmp108_trigger_handle_one_shot;
+
+	/* save this driver instance for passing to other functions */
+	drv_data->tmp108_dev = dev;
+
+#ifdef CONFIG_TMP108_ALERT_INTERRUPTS
+	setup_interrupts(dev);
+#endif
+
+	return 0;
+}
+
+#define TMP108_INST(inst)						   \
+	static struct tmp108_data tmp108_prv_data_##inst;		   \
+	static const struct tmp108_config tmp108_config_##inst = {	 \
+		.i2c_spec = I2C_DT_SPEC_INST_GET(inst),			   \
+		.alert_gpio = GPIO_DT_SPEC_INST_GET(inst, alert_gpios) \
```

Comment:
Since alert-gpios is an optional property, you need to use GPIO_DT_SPEC_INST_GET_OR here instead.